addons/udes_stock/models/product_template.py

```diff
@@ -26,6 +26,37 @@ def _constrain_non_negative_values(self):
                         % (field, record.name)
                     )
 
+    u_pack_qty = fields.Integer(
+        "Quantity Per Pack",
+        default=1,
+        help=_("Quantity of products per pack.")
+    )
+    u_carton_qty = fields.Integer(
+        "Quantity Per Carton",
+        default=1,
+        help=_("Quantity of products per Carton.")
+    )
+    u_pallet_qty = fields.Integer(
+        "Quantity Per Pallet",
+        default=1,
+        help=_("Quantity of products per pallet.")
+    )
+
+    POSITIVE_FIELDS = [
+        "u_pack_qty",
+        "u_carton_qty",
+        "u_pallet_qty",
+    ]
+    @api.constrains(*POSITIVE_FIELDS)
+    def _constrain_positive_values(self):
+        for record in self:
+            for field in record.POSITIVE_FIELDS:
```

Comment:
Does this not give `AttributeError: 'product.template' object has no attribute 'POSITIVE_FIELDS'`?

Reply:
It works fine.